samples/java_layout/src/main/java/io/customer/android/sample/java_layout/ui/common/SimpleFragmentActivity.java

```diff
@@ -38,17 +44,86 @@ public void onBackPressed() {
         finish();
     }
 
+    @Override
+    protected void injectDependencies() {
+        authViewModel = viewModelProvider.get(AuthViewModel.class);
+    }
+
+    private void navigateUp() {
+        // For better user experience, navigate to launcher activity on navigate up button
+        if (isTaskRoot()) {
+            startActivity(new Intent(SimpleFragmentActivity.this, DashboardActivity.class));
+        }
+        onBackPressed();
+    }
+
     @Override
     protected void setupContent() {
-        Bundle extras = getIntent().getExtras();
-        if (extras != null) {
-            mFragmentName = extras.getString(ARG_FRAGMENT_NAME);
+        if (!parseFragmentParams()) {
+            // Navigate up for unsupported deep links so app doesn't crash and user experience is
+            // not affected
+            navigateUp();
+            return;
         }
 
         if (TextUtils.isEmpty(mFragmentName)) {
             throw new IllegalStateException("Fragment name cannot be null");
         }
 
+        binding.topAppBar.setNavigationOnClickListener(view -> navigateUp());
+        authViewModel.getUserLoggedInStateObservable().observe(this, isLoggedIn -> {
+            if (isLoggedIn) {
+                replaceFragment();
+            } else {
+                if (isTaskRoot()) {
+                    startActivity(new Intent(SimpleFragmentActivity.this, LoginActivity.class));
+                }
+                finish();
+            }
+        });
+    }
+
+    private boolean parseFragmentParams() {
+        Intent intent = getIntent();
+        Uri data = intent.getData();
```

Comment:
```suggestion
        Uri deepLinkUrl = intent.getData();
```

Small thing, but I think the code reads better with a more descriptive variable to more quickly recognize what this variable is.